cmd/oras/internal/option/target.go

```diff
@@ -118,24 +118,23 @@ func (opts *Target) Parse() error {
 }
 
 // parseOCILayoutReference parses the raw in format of <path>[:<tag>|@<digest>]
-func parseOCILayoutReference(raw string) (path string, ref string, err error) {
+func (opts *Target) parseOCILayoutReference() (path string, ref string, err error) {
+	raw := opts.RawReference
 	if idx := strings.LastIndex(raw, "@"); idx != -1 {
 		// `digest` found
 		path = raw[:idx]
 		ref = raw[idx+1:]
 	} else {
 		// find `tag`
 		path, ref, err = fileref.Parse(raw, "")
+		if err != nil {
+			err = errors.Join(err, errdef.ErrInvalidReference)
+		}
```

Comment:
Also setup target option for digest code path
```suggestion
		if err != nil {
			err = errors.Join(err, errdef.ErrInvalidReference)
		} else {
		        opts.Path = path
		        opts.Reference = ref
		}
```